### ISSUE TITLE
[BUG] Classes that implement class 'o.o.cluster.support.master.LocalNodeMasterListener' should not have to implement abstract method 'onClusterManager()' and 'offClusterManager()'

### ISSUE THREAD

**Issue by tlfeng** [opensearch-project/OpenSearch#3688]
**Describe the bug**
Similar issue: #3663 and #3683
Introduced by PR https://github.com/opensearch-project/OpenSearch/pull/2519 / commit https://github.com/opensearch-project/OpenSearch/commit/79eb3b04922f4d4688372b93550ded9feffd7338, which aims to resolve issue https://github.com/opensearch-project/OpenSearch/issues/1548

The problem:
2 abstract methods `onMaster()` and `offMaster()` of interface [LocalNodeMasterListener](https://github.com/opensearch-project/OpenSearch/blob/2.0.1/server/src/main/java/org/opensearch/cluster/LocalNodeMasterListener.java) got renamed directly.
It broke the backwards compatibility that all classes that implements the interface has to implement the methods in the new name.

It causes build failure for plugins, such as https://github.com/opensearch-project/index-management/blob/b6e7d35c4fe4b64a41ef9e45d0cafcdfe0289042/src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/IndexStateManagementHistory.kt#L85

A temporary solution is to revert the changes, which means change back the method name `onMaster()` and `offMaster()` in the interface [LocalNodeMasterListener](https://github.com/opensearch-project/OpenSearch/blob/2.0.1/server/src/main/java/org/opensearch/cluster/LocalNodeMasterListener.java).

**To Reproduce**
Build [Index management plugin](https://github.com/opensearch-project/index-management) with OpenSearch 2.1.0 (the code at https://github.com/opensearch-project/OpenSearch/tree/304d830275fdbe5f8095a7285ca66266987c727a)

`java.lang.AbstractMethodError: Receiver class org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementHistory does not define or inherit an implementation of the resolved method 'abstract void onClusterManager()' of interface org.opensearch.cluster.LocalNodeMasterListener.`

**Expected behavior**
Index management plugin should successfully be built with OpenSearch 2.1.0

**Additional context**

**Comment by tlfeng** [comment]
The temporary solution has been applied to `2.1` branch by the PR https://github.com/opensearch-project/OpenSearch/pull/3687 / commit https://github.com/opensearch-project/OpenSearch/commit/8aadcc056513a2edf21978d2c9a2f41756d8366c

**Comment by tlfeng** [comment]
Close the issue. As the `onMaster()` and `offMaster()` method name have been restored in all branches.
PR https://github.com/opensearch-project/OpenSearch/pull/3686 for `main` branch, and PR https://github.com/opensearch-project/OpenSearch/pull/3693 for `2.x` branch.

**Comment by tlfeng** [comment]
A new unit test is added (https://github.com/opensearch-project/OpenSearch/blob/0e96a878a5ab1ffec3ee04f15fe976d34c9905dc/server/src/test/java/org/opensearch/cluster/service/ClusterApplierServiceTests.java#L340) to validate the backwards compatibility for the abstract methods `onMaster()` and `offMaster()`.

**Comment by tlfeng** [comment]
Specifically, the solution provided by @andrross  for deprecating the abstract methods in interface `LocalNodeMasterListner` is:
```
@deprecated
public interface LocalNodeMasterListener extends LocalNodeClusterManagerListener {
    void onMaster();

    void offMaster();

    @Override
    default void onClusterManager() {
        onMaster()
    }

    @Override
    default void offClusterManager() {
        offMaster();
    }
}

public interface LocalNodeClusterManagerListener extends ClusterStateListener {
    void onClusterManager();

    void offClusterManager();
}
```
> All places that take a `LocalNodeMasterListener` instance can be changed to take a `LocalNodeClusterManagerListener`. Developers can still use `LocalNodeMasterListener` everywhere and it will still work as-is, though they’ll get deprecation warnings.

**Comment by tlfeng** [comment]
The final solution in PR https://github.com/opensearch-project/OpenSearch/pull/4121 has been merged, and from version 2.2.0, developers can use `onClusterManager()` and `offClusterManager()` to replace `onMaster()` and `offMaster()`
For more detail: https://github.com/opensearch-project/OpenSearch/issues/3543#issuecomment-1204744584
